cmd/earthly/main.go

```diff
@@ -1363,7 +1363,7 @@ func (app *earthlyApp) actionBootstrap(c *cli.Context) error {
 	var err error
 	console := app.console.WithPrefix("bootstrap")
 
-	if app.bootstrapWithAutocomplete {
+	if app.bootstrapWithAutocomplete || app.homebrewSource != "" {
```

Comment:
I think these shuold be two separate if blocks actually.

Reply:
Basically, the switch should be completely independent from `bootstrapWithAutocomplete`.

And `bootstrapWithAutocomplete` should only wrap the stuff that comes after the switch.

Reply:
Currently homebrew calls:

    "#{bin}/earthly", "bootstrap", "--source", "bash"

(see https://github.com/Homebrew/homebrew-core/blob/master/Formula/earthly.rb#L33 )

which should print out something like this:

    complete -o nospace -C '/home/alex/bin/earthly-v0.5.8' earthly

(or a different path depending on the system).

If this block was broken up into a seperate part, maybe we can remove the need to change how homebrew generates this output?

Reply:
Yep. The switch above actually returns if it was a `--source` call. I had just skimmed it and though it was getting the proper data for autocompletion.

I gated the switch separate from the other stuff, and then also needed to move the permissions insurance to a defer. See the new fix and if this gets us there?